Source/Csla/Core/GraphMerger.cs

```diff
@@ -129,16 +129,27 @@ private static void CopyField(object source, object target, string fieldName)
     /// </summary>
     /// <param name="target">Target of merge.</param>
     /// <param name="source">Source for merge.</param>
-    private void MergeGraph(IEditableCollection target, IEditableCollection source)
+    /// <param name="isAsync">Called from sync or async method.</param>
+    private void MergeGraph(IEditableCollection target, IEditableCollection source, bool isAsync = false)
     {
       var listType = target.GetType();
       var childType = Utilities.GetChildItemType(listType);
       var genericTypeParams = new Type[] { listType, childType };
       System.Reflection.MethodInfo methodReference;
-      if (typeof(IExtendedBindingList).IsAssignableFrom(listType))
-        methodReference = GetType().GetMethod("MergeBusinessBindingListGraph");
+      if(isAsync)
+      {
+        if (typeof(IExtendedBindingList).IsAssignableFrom(listType))
+          methodReference = GetType().GetMethod("MergeBusinessBindingListGraphAsync");
```

Comment:
I see where you find the methods ending with `Async`, but not where you are invoking them in an async manner? Are they still being invoked as sync methods?